x/stakeibc/keeper/unbonding.go

```diff
@@ -485,8 +482,11 @@ func (k Keeper) UnbondFromHostZone(ctx sdk.Context, hostZone types.HostZone) err
 	}
 
 	// Send the messages in batches so the gas limit isn't exceedeed
-	for start := 0; start < len(msgs); start += UndelegateICABatchSize {
-		end := start + UndelegateICABatchSize
+	// NOTE: With the current implementation, an error is thrown upstream if there are
+	// more messages than undelegateBatchSize - so there should only be one iteration in this loop
+	// This is because the undelegation callback is not currently setup to handle multiple batches
+	for start := 0; start < len(msgs); start += undelegateBatchSize {
```

Comment:
I know we check upstream, but do you think it's at all worth it to add a redundant check here to verify this loop only fires once? This is just one of the places that's the most risky if anything unexpected happened

Reply:
imo, I don't think it's necessary. At that point, I think we'd just remove the loop, but sounded like we were on board with laying off this code for now to make the change less invasive

Reply:
Makes sense, I also didn't realize the check is run so close to here (line 468). On-board with this, resolving